src/control/client/pool.go

```diff
@@ -258,3 +258,48 @@ func (c *connList) PoolUpdateACL(req *PoolUpdateACLReq) (*PoolUpdateACLResp, err
 		ACL: &AccessControlList{Entries: pbResp.ACL},
 	}, nil
 }
+
+// PoolDeleteACLReq contains the input parameters for PoolDeleteACL.
+type PoolDeleteACLReq struct {
+	UUID      string // UUID of the pool
+	Principal string // Principal whose entry will be removed
+}
+
+// PoolDeleteACLResp returns the updated ACL for the pool.
+type PoolDeleteACLResp struct {
+	ACL *AccessControlList // actual ACL of the pool
+}
+
+// PoolDeleteACL sends a request to delete an entry in a pool's Access Control
+// List. If it succeeds, it returns the updated ACL. If it fails, it returns an
+// error.
+func (c *connList) PoolDeleteACL(req *PoolDeleteACLReq) (*PoolDeleteACLResp, error) {
```

Comment:
I didn't notice this to call it out on previous reviews in this area, but one pattern I've been using for these kinds of req/res API methods is to accept a struct and return a pointer. This ensures that you can't get a nil request. If there's a good reason for accepting a pointer (usually not, side effects are bad mmkay), then your method should at least check to make sure the request isn't nil before starting to access stuff in it.

Reply:
Good catch. I can clean these up in a follow-on patch, or in this one if there are any changes requested by others. Thanks.